cmd/gke-gcloud-auth-plugin/gcloud_token_provider.go

```diff
@@ -87,6 +90,27 @@ func (p *gcloudTokenProvider) useCache() bool {
 	return true
 }
 
-func readGcloudConfigRaw() ([]byte, error) {
-	return executeCommand("gcloud", "config", "config-helper", "--format=json")
+func (p *gcloudTokenProvider) getExtraArgs() []string {
+	extraArgs := make([]string, 0, 3)
+	if p.project != "" {
+		extraArgs = append(extraArgs, fmt.Sprintf("--project=%s", p.project))
+	}
+	if p.account != "" {
+		extraArgs = append(extraArgs, fmt.Sprintf("--account=%s", p.account))
+	}
+	if p.impersonateServiceAccount != "" {
+		extraArgs = append(extraArgs, fmt.Sprintf("--impersonate-service-account=%s", p.impersonateServiceAccount))
+	}
+	return extraArgs
+}
+
+func readGcloudConfigRaw(extraArgs ...string) ([]byte, error) {
+	args := []string{
+		"config",
+		"config-helper",
+		"--format=json",
+	}
+	args = append(args, extraArgs...)
+	klog.V(4).Infof("Executing gcloud command with args: %v", args)
+	return executeCommand("gcloud", args...)
```

Comment:
It would be great if we could parameterize executeCommand as p.executeCommand and add unit test to ensure commands are getting created correctly when it reaches execute command.